packages/jupyter-ai/src/components/chat-messages.tsx

```diff
@@ -106,24 +105,32 @@ export function ChatMessages(props: ChatMessagesProps) {
   const [timestamps, setTimestamps] = useState<Record<string, string>>({});
 
   /**
-   * Effect: update cached timestamp strings upon receiving a new message and
-   * every 5 seconds after that.
+   * Effect: update cached timestamp strings upon receiving a new message.
    */
   useEffect(() => {
     function updateTimestamps() {
       const newTimestamps: Record<string, string> = {};
+      let timestampAdded: boolean = false;
+
       for (const message of props.messages) {
-        newTimestamps[message.id] =
-          formatDistanceToNowStrict(fromUnixTime(message.time)) + ' ago';
+        if (!(message.id in newTimestamps)) {
+          // Use the browser's default locale
+          newTimestamps[message.id] =
+            new Date(message.time * 1000) // Convert message time to milliseconds
+              .toLocaleTimeString([], {
+                hour: 'numeric', // Avoid leading zero for hours; we don't want "03:15 PM"
+                minute: '2-digit'
+              });
+
+          timestampAdded = true;
+        }
+      }
+      if (timestampAdded) {
+        setTimestamps(newTimestamps);
       }
-      setTimestamps(newTimestamps);
     }
 
     updateTimestamps();
```

Comment:
You can unwrap the `updateTimestamps()` function since it's only being called once in this effect hook.